server/src/main/java/io/crate/replication/logical/ShardReplicationService.java

```diff
@@ -121,7 +121,7 @@ public void beforeIndexShardClosed(ShardId shardId,
     }
 
     @Override
-    public void afterIndexShardDeleted(ShardId shardId, Settings indexSettings) {
+    public void beforeIndexShardDeleted(ShardId shardId, Settings indexSettings) {
```

Comment:
I don't know if there was a good reason to have this on `after` ?

Reply:
Can't really remember, sorry. Maybe no real reason for it. Doing it `before` looks fine. What is the reason to change it? higher chance that the tracker (or request) actual stops when the shard is gone?